modules/custom/d_content/modules/d_content_init/src/ContentInitManagerBase.php

```diff
@@ -0,0 +1,187 @@
+  }
+
+  /**
+   * Import content from YML files.
+   *
+   * @param array $structure
+   *   Structure of the content to import.
+   */
+  public function importFromFiles(array $structure) {
+    foreach ($structure as $data) {
+      $this->importFromFile($data['file']);
+    }
+  }
+
+  /**
+   * Import content from specified YML file.
+   *
+   * @param string $file
+   *   Path of the YML file to import.
+   *
+   * @return mixed|bool
+   */
+  protected function importFromFile($file) {
+    $content = $this->getDecodedYmlFile($file);
+    if (!$content) {
+      $this->logger->error($this->t('Decoding of the YML file failed: @file', [
+        '@file' => $file,
+      ]));
+      return FALSE;
+    }
+
+    $create = $this->getCreateMethodName($content['info']['type']);
+    if (!method_exists($this, $create)) {
+      $this->logger->error($this->t('Missing method: @method', [
+        '@method' => $create,
+      ]));
+      return FALSE;
+    }
+    return $this->$create($content);
+  }
+
+  /**
+   * Create and save entity of specified type.
+   *
+   * @param string $entity_type
+   *   Type of the entity to create.
+   * @param array $values
+   *   (optional) An array of values to set, keyed by property name. If the
+   *   entity type has bundles, the bundle key has to be specified.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function saveEntity($entity_type, array $values) {
+    $storage = \Drupal::entityTypeManager()->getStorage($entity_type);
+    $entity = $storage->create($values);
+    $entity->save();
+    return $entity;
+  }
+
+  /**
+   * Process array of fields on the specified entity.
+   *
+   * @param array $block
+   *   Block definition to process.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity to operate on.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function processFields(array $block, EntityInterface &$entity) {
+    if (isset($block['fields'])) {
+      foreach ($block['fields'] as $field_name => $field) {
+        $this->processField($field_name, $field, $entity);
+      }
+      $entity->save();
+    }
+  }
+
+  /**
+   * Process field on the specified entity.
+   *
+   * @param string $field_name
+   *   Field name to process.
+   * @param array $field
+   *   Data to save into field.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity to operate on.
+   *
+   * @return bool|\Drupal\Core\Entity\EntityInterface
+   *   Updated entity on success, FALSE otherwise.
+   */
+  protected function processField($field_name, array $field, EntityInterface &$entity) {
+    if (!$entity->hasField($field_name)) {
+      // @todo: log list of the missing fields?
```

Comment:
Yeah, we can log it, later we'll be able to check messages from `d_content_init` module in Codeception